### PR TITLE
Species outfits are now handled in SSjobs instead of species datums.

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
Moves all outfit data out of species/proc/before_equip_job() and into dictionaries on job datums.

## Why It's Good For The Game
MUCH cleaner.

## Changelog
:cl:
code: Cleans up species outfit code.
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -190,6 +193,12 @@
 		H.account_id = bank_account.account_id
 
 	//Equip the rest of the gear
+
+	if(src.species_outfits)
+		if(H.dna.species.id in src.species_outfits)
+			var/datum/outfit/O = species_outfits[H.dna.species.id]
+			H.equipOutfit(O, visualsOnly)
```
Comment (francinum) [inline code/modules/jobs/job_types/_job.dm:201]: You should probably be setting `outfit_override` (a local variable passed into the proc, check to make sure it's null before checking species stuff too!) instead of equipping.

Comment (Kapu1178) [reply code/modules/jobs/job_types/_job.dm:201]: Species outfits are a equipped before normal job outfits, and, unlike Bay, outfit.equip doesn't override existing clothing items by default. 

I also looked at outfit_override's purpose, and i literally couldn't find one that is relevant to this code.

Tl;dr, it's fine as is.

**Comment by Kapu1178** [comment]
also remove the golden label and put it on grod. s m h

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -190,6 +193,12 @@
...
+
 	H.dna.species.before_equip_job(src, H, visualsOnly)
```
Comment (chloecat0) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
	H.dna.species.before_equip_job(src, H, visualsOnly)
	
	if(src.species_outfits)
		if(H.dna.species.id in src.species_outfits)
			var/datum/outfit/O = species_outfits[H.dna.species.id]
			H.equipOutfit(O, visualsOnly)
```

File: code/modules/mob/living/carbon/human/species_types/plasmamen.dm [hunk 3]
```diff
@@ -61,115 +61,9 @@
 	. = ..()
 
 /datum/species/plasmaman/before_equip_job(datum/job/J, mob/living/carbon/human/H, visualsOnly = FALSE)
```
Comment (chloecat0) [inline code/modules/mob/living/carbon/human/species_types/plasmamen.dm]:
```suggestion
/datum/species/plasmaman/after_equip_job(datum/job/J, mob/living/carbon/human/H, visualsOnly = FALSE)
```

**Review by chloecat0 — COMMENTED** [review]
I feel like before_equip_job should be done before equipping the outfit and plasmaman internals just dealt with in after_equip_job.

File: code/modules/jobs/job_types/_job.dm (hunk 2 above)
Comment (Kapu1178) [reply code/modules/jobs/job_types/_job.dm]: This is probably a good idea?

File: code/modules/mob/living/carbon/human/species_types/plasmamen.dm (hunk 3 above)
Comment (Kapu1178) [reply code/modules/mob/living/carbon/human/species_types/plasmamen.dm]: This is also, a good idea. Probably.

**Comment by Kapu1178** [comment]
Tested webedits. Works fine.

**Comment by BrumalLake** [comment]
point of grod species outfits with spritesheets?

**Comment by Kapu1178** [comment]
> point of grod species outfits with spritesheets?

sprite sheets dynamically adjusts icon states. Outfit Datums handle adding *completely different items* to a job’s equipment.
